src/components/FetchHandler.tsx

```diff
@@ -1,28 +1,40 @@
 import * as React from 'react';
-import { usePromise } from '../hooks/use-promise';
-import { Spinner, Alert } from '../common/components';
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
+import { usePromise } from '../hooks';
+import { Spinner, Alert } from '.';
+
+/**
+ * @param data This is a render-prop which you use to only render your content when data has come into effect
+ * @param setDependencyList This is a callback which you use to set the list of dependencies for the useEffect hook.
+ * Using this will allow you to re-fetch the data if you have a dependency on a submit for example.
+ */
+type Child<TResult> = (data: TResult, setDependencyList: (dependencies: any) => void) => React.ReactNode;
```

Comment:
Kanske borde typa `dependencies: any` som `any[]` och hantera det som en array nedan (genom att ha `[]` som default-värde och spread:a arrayen i andra argumentet till `useEffect`)?

Om man inte gör det, och användaren av `FetchHandler` skickar in en array som dependencies, så resulterar det i en array av arrays i useEffect-dependencies-argumentet, vilket jag antar kan ställa till med buggar iom shallow comparison.